### ISSUE TITLE
Multiple `#define` for the same keywords in SDL_oldnames.h

### ISSUE THREAD

**Issue by Semphriss** [libsdl-org/SDL#9306]
Some keywords, for example `SDL_RWFromFile`, have multiple incompatible redifinitions:

```
include/SDL3/SDL_oldnames.h:461:#define SDL_RWFromFile SDL_IOFromFile
include/SDL3/SDL_oldnames.h:962:#define SDL_RWFromFile SDL_RWFromFile_renamed_SDL_IOFromFile
```

I'm porting some software to SDL3 and I've been getting warnings like this:

```
warning: initialization of ‘SDL_IOStream *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
```

... which made me confused for a while.

I noted that I need to comment both macro definitions for GCC to properly report that `SDL_RWFromFile` does not exist and will be implicitly declared as `int SDL_RWFromFile()`.

**Comment by slouken** [comment]
You shouldn't hit both `#define`s in SDL_oldnames.h, only one of them. If you define `SDL_ENABLE_OLD_NAMES` then the renamed functions should mostly just work. If you don't, you should get an error message that `SDL_RWFromFile_renamed_SDL_IOFromFile` does not exist, which should help you with renaming.

What's happening here that's preventing that from working in your case?

**Comment by Semphriss** [comment]
The issue is just that GCC 11.4.0 fails to detect the non-existent function as such, and silently presumes some function with the given name exists with the default signature `int()`. I presumed that the mechanism was meant to make the compiler fail at compilation and prominently show the notice-like identifier.

But after further use, I found that the linker will eventually complain that `SDL_foo_renamed_SDL_bar` does not exist, although it does not report the file/line where the problem lives.

**Comment by sezero** [comment]
> The issue is just that GCC 11.4.0 fails to detect the non-existent function 

That's the intention: it should result in link failure

**Comment by Semphriss** [comment]
> > The issue is just that GCC 11.4.0 fails to detect the non-existent function
> 
> That's the intention: it should result in link failure

I'm sorry, I just realised I worded my message poorly. What I originally meant was that GCC will fail to detect the _non-existence_ of the function; I meant that GCC will happily compile in the non-existing functions as if they existed and presume that they're just declared elsewhere, which is a legal thing to do in C, and won't realise there's a problem before the linking stage. Since the `#define` results in GCC not printing any useful warning about the silent "presumed implicit declaration" for some reason, and a compilation failure says in which file and at which line the problematic call is, but not a linking failure, I thought that was an accident.

However, I'm not sure if it's possible to get GCC to print an error (or at least a more explicit warning) by default at compilation. There's probably a `-Werror=` for that, but I'll try to look if there's a more reliable way to do this. Maybe this is just a quirk of C/GCC, but I'll keep you updated if I find anything.

**Comment by sezero** [comment]
`-Werror=strict-prototypes` (or something like it) should help

**Comment by Semphriss** [comment]
`-Werror=implicit-function-declaration` is what I was looking for, but then I realised that I've installed SDL3 on my system manually, and GCC automatically ignores warnings in system header files (including, apparently,  those that are `#define`'d). So I'd have to get GCC to show errors in system header files, or to configure the include directory for SDL as a non-system header. In any case, it doesn't look like anything can be done on the SDL side.
